compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/align-scopes-trycatch-nested-overlapping-range.expect.md

```diff
@@ -0,0 +1,51 @@
+
+## Input
+
+```javascript
+import { CONST_TRUE, makeObject_Primitives } from "shared-runtime";
+
+function Foo() {
+  try {
+    let thing = null;
+    if (cond) {
+      thing = makeObject_Primitives();
+    }
+    if (CONST_TRUE) {
+      mutate(thing);
+    }
+  } catch {}
+}
+
+export const FIXTURE_ENTRYPOINT = {
+  fn: Foo,
+  params: [{}],
+};
+
+```
+
+## Code
+
+```javascript
+import { CONST_TRUE, makeObject_Primitives } from "shared-runtime";
+
+function Foo() {
+  try {
+    let thing = null;
+    if (cond) {
+      thing = makeObject_Primitives();
```

Comment:
just curious, why do we lose memoization on this?

Reply:
Ahh memoization was pruned because we didn't return `item`. Just updated the fixture to return `item` to avoid pruning!